src/sysinfo/nvapi_adapter_registry.h

```diff
@@ -30,8 +31,12 @@ namespace dxvk {
         [[nodiscard]] NvapiOutput* FindPrimaryOutput() const;
         [[nodiscard]] bool IsOutput(NvapiOutput* handle) const;
 
+        IDXGIVkInteropFactory1 *GetInteropFactory() const { return m_dxgiVkInterop.ptr(); }
+
       private:
         ResourceFactory& m_resourceFactory;
+        Com<IDXGIFactory1> m_dxgiFactory;
+        Com<IDXGIVkInteropFactory1> m_dxgiVkInterop;
```

Comment:
IIRC there were previously some concerns about keeping DXGI factories and/or Vulkan instances alive for the entire lifetime of NVAPI but I don't see an easy way around this now.

@jp7677 I hope this is fine.

Reply:
Yes, keeping the dxgi factory alive is imho also the way to go. Creating a new dxgi factory on every hdr-color-control call would be an alternative, but certainly not a good one since DXVK does some initialization work in the dxgi factory constructor.

Once this PR has landed, we can probably revisit/refactor the earlier life time changes to make things simpler again.

Reply:
dxvk nvapi creates the factory so that wouldn't work, we don't use a factory passed in from the app.